### PR TITLE
S2 optical prop

### PR DESCRIPTION
**Pull request by KaraMelih**
**What is the problem / what does the code in this PR do**
Samples optical propagation delays in the gas phase from a G4 simulated data.
The S2 pulse workflow is a bit changed; photon channels sampling is now independent from photon times. This was needed so that photon time delays can be added based on whether they are detected at top/bottom arrays.
!To add this effect it uses a local aux file for now, the file will be added to private_nt_aux repo!

**WFSim needs you:**
 - Please add a test for this PR, as a bare minimum, make sure it's covered in coveralls!
 - Can you add a docsting to all your functions?

*Pay attention:*
 - Due to databases being needed for testing, making a PR from your own fork will typically NOT run the tests. If you then merge master might break

### CODE REVIEW

**Comment by JYangQi00** [comment]
Hi Melih, this looks good, but how do you propose we change the config to include 'confxy'?

File: wfsim/core/s2.py [hunk 1]
```diff
@@ -332,42 +330,96 @@ def photon_timings(t, n_electron, z, xy, sc_gain, config, resource, phase):
...
+        :param config: dict of the wfsim config
+        :param resource: instance of the resource class
+        """
+        # Luminescence Timings
+        if 'simple' in config['s2_luminescence_model']:
+            photon_timings = S2.luminescence_timings_simple(xy, n_photons_per_xy,
+                                                             config=config,
+                                                             resource=resource)
+        elif 'garfield' in config['s2_luminescence_model']:
+            confxy = True if 'confine distance' in config['s2_luminescence_model'] else False
```
Comment (KaraMelih) [inline wfsim/core/s2.py:390]: @JYangQi00 Hi Jianyang currently, I added it here. So while comparing the two, I simulate once setting `{'s2_luminescence_model':'garfield'}` and once with `{'s2_luminescence_model':'garfield confine distance'}` so both `confxy=True` and `confxy=False` are callable

Comment (KaraMelih) [reply wfsim/core/s2.py]: and [here](https://github.com/XENONnT/analysiscode/blob/master/SE_WFSim_shape/data-WFSim_comparison.ipynb) I uploaded a notebook, using this updates. I'm trying to compare the data vs WFSim with several configs

**Review by JYangQi00 — APPROVED** [review]
Okay, I see, looks good!
